src/main/java/net/minecraftforge/common/loot/LootModifier.java

```diff
@@ -0,0 +1,105 @@
+package net.minecraftforge.common.loot;
+
+import java.util.List;
+import java.util.function.Predicate;
+
+import javax.annotation.Nonnull;
+
+import com.google.gson.JsonObject;
+
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.world.storage.loot.LootContext;
+import net.minecraft.world.storage.loot.conditions.ILootCondition;
+import net.minecraft.world.storage.loot.conditions.LootConditionManager;
+import net.minecraftforge.registries.GameData;
+
+/**
+ * A base implementation of a Global Loot Modifier for modders to extend.
+ * Takes care of ILootCondition matching and comes with a base serializer
+ * implementation that takes care of Forge registry things.
+ */
+public abstract class LootModifier implements IGlobalLootModifier {
+    protected final ILootCondition[] conditions;
+    private final Predicate<LootContext> combinedConditions;
+    
+    /**
+     * Constructs a LootModifier.
+     * @param conditionsIn the ILootConditions that need to be matched before the loot is modified.
+     */
+    protected LootModifier(ILootCondition[] conditionsIn) {
+        this.conditions = conditionsIn;
+        this.combinedConditions = LootConditionManager.and(conditionsIn);
+    }
+    
+    @Nonnull
+    @Override
+    public final List<ItemStack> apply(List<ItemStack> generatedLoot, LootContext context) {
+        return this.combinedConditions.test(context) ? this.doApply(generatedLoot, context) : generatedLoot;
+    }
+    
+    /**
+     * Applies the modifier to the generated loot (all loot conditions have already been checked
+     * and have returned true).
+     * @param generatedLoot the list of ItemStacks that will be dropped, generated by loot tables
+     * @param context the LootContext, identical to what is passed to loot tables
+     * @return modified loot drops
+     */
+    @Nonnull
+    protected abstract List<ItemStack> doApply(List<ItemStack> generatedLoot, LootContext context);
+    
+    /**
+     * Abstract base deserializer for LootModifiers. Takes care of Forge registry things.<br/>
+     * Modders should extend this class to return their modifier and implement the abstract
+     * <code>read</code> method to deserialize from json.
+     * @param <T> the final Type
+     */
+    public abstract static class Serializer<T extends LootModifier> implements IGlobalLootModifierSerializer<T> {
```

Comment:
Shouldn't this be extending `ForgeRegistryEntry<IGlobalLootModifierSerializer<?>>` instead of implementing `IForgeRegsitryEntry<T>` by hand?

Actually, I wonder if it wouldn't be best to make `IGlobalLootModifierSerializer` an abstract class instead, then subclasses wouldn't need to worry about IForgeRegsitryEntry

Reply:
Hmm, cursory check shows that that is doable (it ran and didn't crash, can't check more than that at this instant).

Reply:
Yep, it works fine that way (had just wanted to verify that things were still being loaded and the test mod working). I personally don't care if its an interface or an abstract class, but it does seem cleaner/easier on the modder as the abstract class.

Reply:
The main reason to use an interface over an abstract class, is that you don't force users into a specific implementation, since they can implement that interface anywhere they like.
However I'd argue that in this case, there isn't much point of keeping it an interface? People shouldn't be making any random object their loot modifier serializer.

Reply:
Yep, agreed. There was nothing in the abstract implementation that a modder needed to touch, so I pushed it upwards and made it final.